pkg/apis/eventing/v1alpha1/broker_validation_test.go

```diff
@@ -38,11 +39,58 @@ func TestBrokerSpecValidation(t *testing.T) {
 	_ = bs.Validate(context.TODO())
 }
 
-// No-op test because method does nothing.
+type noBroker struct{}
+
+func (nb noBroker) CheckImmutableFields(_ context.Context, _ apis.Immutable) *apis.FieldError {
+	return nil
+}
+
 func TestBrokerImmutableFields(t *testing.T) {
-	original := &Broker{}
-	current := &Broker{}
-	_ = current.CheckImmutableFields(context.TODO(), original)
+	original := &Broker{
+		Spec: BrokerSpec{
+			ChannelTemplate: &eventingduckv1alpha1.ChannelTemplateSpec{TypeMeta: metav1.TypeMeta{Kind: "my-kind"}},
+		},
+	}
+	current := &Broker{
+		Spec: BrokerSpec{
+			ChannelTemplate: &eventingduckv1alpha1.ChannelTemplateSpec{TypeMeta: metav1.TypeMeta{Kind: "my-other-kind"}},
+		},
+	}
+	diff, err := kmp.ShortDiff(original.Spec.ChannelTemplate, current.Spec.ChannelTemplate)
+	if err != nil {
+		t.Fatalf("failed to diff current and origional Broker ChannelTemplate: %v", err)
+	}
+
+	tests := []struct {
+		name    string
+		og      apis.Immutable
+		wantErr *apis.FieldError
+	}{{
+		name:    "invalid original",
+		og:      &noBroker{},
+		wantErr: &apis.FieldError{Message: "The provided original was not a Broker"},
+	}, {
+		name:    "no ChannelTemplateSpec mutation",
+		og:      current,
+		wantErr: nil,
+	}, {
+		name: "ChannelTemplateSpec mutated",
+		og:   original,
+		wantErr: &apis.FieldError{
+			Message: "Immutable fields changed (-old +new)",
+			Paths:   []string{"spec", "channelTemplate"},
+			Details: diff,
+		},
+	}}
+
+	for _, test := range tests {
```

Comment:
Up to you if want to do this or not, as the repo uses both styles:

My preference is to have tests be a `map[string]struct { ... }`. The key being the name of the test case. Then this line would become 
```suggestion
	for n, test := range tests {
```

Reply:
Interesting (possible useful) consequence of using a map instead of an array: the order of test evaluation is non-deterministic.